### PR TITLE
fix implement of aac and enable it.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- modules/SceAudiodecUser: enable AAC codec.
- codec/aac: fix aac implement.
- enable lle of mp4 and avplayer in auto mode.
- delete obsolete setting.

# Result:
- fix lbp video
image

- tales of serie
image

- wipeout
image

- PSBAR
image

and more....

### CODE REVIEW

File: vita3k/codec/src/aac.cpp [hunk 1]
```diff
@@ -129,29 +82,18 @@ bool AacDecoderState::receive(uint8_t *data, DecoderSize *size) {
...
-            this, // user data
-            [](void *user, uint8_t *data, int size) -> int {
-                auto *decoder = reinterpret_cast<AacDecoderState *>(user);
-
-                return decoder->read_packet_cache(data, size);
-            }, // read
-            nullptr,
-            nullptr),
-        &avio_close);
+    // io_buffer = std::shared_ptr<uint8_t>(reinterpret_cast<uint8_t *>(av_malloc(buffer_size)), &av_free);
```
Comment (bookmist) [inline vita3k/codec/src/aac.cpp:85]: This two lines are unused. They can be deleted.

**Comment by bookmist** [comment]
I would split this on two commits: one for mspace and other for aac codec

**Comment by Zangetsu38** [comment]
> I would split this on two commits: one for mspace and other for aac codec

Yes not worry, I already planned to do that.
I used one commit juste for more easy set change, i split it when is finished.

File: vita3k/codec/src/aac.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/codec/src/aac.cpp:85]: Oki 👌

Comment (Zangetsu38) [reply vita3k/codec/src/aac.cpp]: done

File: vita3k/codec/src/aac.cpp [hunk 2]
```diff
@@ -129,29 +82,17 @@ bool AacDecoderState::receive(uint8_t *data, DecoderSize *size) {
...
     AVCodec *codec = avcodec_find_decoder(AV_CODEC_ID_AAC);
     assert(codec);
 
     context = avcodec_alloc_context3(codec);
     assert(context);
 
+    context->codec_type = AVMEDIA_TYPE_AUDIO;
     context->channels = channels;
     context->sample_rate = sample_rate;
+    context->channel_layout = AV_CH_LAYOUT_STEREO;
```
Comment (bookmist) [inline vita3k/codec/src/aac.cpp:95]: switch (channels) {
    case 1: context->channel_layout = AV_CH_LAYOUT_MONO;
    case 2: context->channel_layout = AV_CH_LAYOUT_STEREO;
    }

File: vita3k/codec/include/codec/state.h [hunk 3]
```diff
@@ -173,24 +176,21 @@ struct PCMDecoderState : public DecoderState {
 struct AacDecoderState : public DecoderState {
     std::shared_ptr<uint8_t> io_buffer;
     std::shared_ptr<AVIOContext> io_context;
```
Comment (bookmist) [inline vita3k/codec/include/codec/state.h:178]: Not sure, but this two fields are not used

Comment (Zangetsu38) [reply vita3k/codec/include/codec/state.h:178]: ahh yes lol

File: vita3k/codec/src/aac.cpp [hunk 4]
```diff
@@ -63,96 +38,69 @@ uint32_t AacDecoderState::get(DecoderQuery query) {
...
-    av_frame_free(&frame);
     return true;
 }
 
+uint32_t AacDecoderState::get_es_size() {
+    return es_size_used;
+}
+
 AacDecoderState::AacDecoderState(uint32_t sample_rate, uint32_t channels) {
     constexpr size_t buffer_size = 8192;
```
Comment (bookmist) [inline vita3k/codec/src/aac.cpp:85]: not used

Comment (Zangetsu38) [reply vita3k/codec/src/aac.cpp:85]: is used

File: vita3k/codec/src/aac.cpp [hunk 5]
```diff
@@ -63,96 +38,70 @@ uint32_t AacDecoderState::get(DecoderQuery query) {
...
+    int got_frame;
+    int len = codec->decode(context, frame, &got_frame, packet);
+    assert(got_frame);
 
-    err = avcodec_send_packet(context, packet);
     av_packet_free(&packet);
-    if (err < 0) {
-        LOG_WARN("Error sending Aac packet: {}.", codec_error_name(err));
+    if (len < 0) {
+        LOG_WARN("Error sending Aac packet.");
```
[the remark this answers is not in the document]
Comment (Macdu) [reply vita3k/codec/src/aac.cpp:53]: Yes, indeed, I thought what was returned by codec->decode was not an error if it failed but it is.

File: vita3k/codec/src/aac.cpp [hunk 6]
```diff
@@ -63,96 +38,70 @@ uint32_t AacDecoderState::get(DecoderQuery query) {
...
-            frame->sample_rate);
+            context->channels, frame->nb_samples,
+            context->sample_rate);
     }
 
     if (size) {
         size->samples = frame->nb_samples;
     }
 
-    av_frame_free(&frame);
```
[the remark this answers is not in the document]
Comment (Macdu) [reply vita3k/codec/src/aac.cpp:126]: Both are the same, maybe this one is slightly better if for whatever reason you want to receive multiple times.

File: vita3k/codec/src/aac.cpp (hunk 4 above)
Comment (Macdu) [reply vita3k/codec/src/aac.cpp:85]: Used and necessary here https://github.com/Zangetsu38/Vita3K/blob/a5a45e967275fe976890c3680c8a691daef902d6/vita3k/modules/SceAudiodec/SceAudiodecUser.cpp#L207

Comment (bookmist) [reply vita3k/codec/src/aac.cpp:85]: I comment out 
constexpr size_t buffer_size = 8192;
and got no errors

Comment (Macdu) [reply vita3k/codec/src/aac.cpp:85]: You highlighted get_es_size, but otherwise I agree that buffer_size is useless.

Comment (Zangetsu38) [reply vita3k/codec/src/aac.cpp:85]: ah yes buffer_size is not used, you ping on get es size line hdre lol

**Review by bookmist — APPROVED** [review]

**Review by bookmist — APPROVED** [review]

**Pull request merged**